src/data/list/min_max.lean

```diff
@@ -1,176 +1,256 @@
 /-
 Copyright (c) 2019 Minchao Wu. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Author: Minchao Wu
+Authors: Minchao Wu, Chris Hughes
 -/
-import data.list algebra.order_functions
+import data.list.basic
+/-
+# Minimum and maximum of lists
+
+## Main definitions
+
+The main definition are `argmax`, `argmin`, `minimum` and `maximum` for lists.
 
+`argmax f l` returns `some a`, where `a` of `l` that maximises `f a`. If there are `a b` such that
+  `f a = f b`, it returns whichever of `a` or `b` comes first in the list.
+  `argmax f []` = none`
+
+`minimum l` returns the smallest element of `l`. `minimum [] = none`
```

Comment:
I'd use `with_top`/`with_bot` instead of `option` here.

Reply:
Done